tests/by-util/test_install.rs

```diff
@@ -333,3 +336,129 @@ fn test_install_target_file_dev_null() {
     ucmd.arg(file1).arg(file2).succeeds().no_stderr();
     assert!(at.file_exists(file2));
 }
+
+#[test]
+fn test_install_copy_then_compare_file() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let file1 = "test_install_copy_then_compare_file_a1";
+    let file2 = "test_install_copy_then_compare_file_a2";
+
+    at.touch(file1);
+    scene
+        .ucmd()
+        .arg("-C")
+        .arg(file1)
+        .arg(file2)
+        .succeeds()
+        .no_stderr();
+
+    let mut file2_meta = at.metadata(file2);
+    let before = FileTime::from_last_modification_time(&file2_meta);
+
+    scene
+        .ucmd()
+        .arg("-C")
+        .arg(file1)
+        .arg(file2)
+        .succeeds()
+        .no_stderr();
+
+    file2_meta = at.metadata(file2);
+    let after = FileTime::from_last_modification_time(&file2_meta);
+
+    assert!(before == after);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_install_copy_then_compare_file_with_extra_mode() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    // XXX: can't tests introspect on their own names?
+    let file1 = "test_install_copy_then_compare_file_with_extra_mode_a1";
+    let file2 = "test_install_copy_then_compare_file_with_extra_mode_a2";
+
+    at.touch(file1);
+    scene
+        .ucmd()
+        .arg("-C")
+        .arg(file1)
+        .arg(file2)
+        .succeeds()
+        .no_stderr();
+
+    let mut file2_meta = at.metadata(file2);
+    let before = FileTime::from_last_modification_time(&file2_meta);
+
+    scene
+        .ucmd()
+        .arg("-C")
+        .arg(file1)
+        .arg(file2)
+        .arg("-m")
+        .arg("1644")
+        .succeeds()
+        .no_stderr();
+
+    file2_meta = at.metadata(file2);
+    let after_install_sticky = FileTime::from_last_modification_time(&file2_meta);
+
+    assert!(before != after_install_sticky);
+
+    // dest file still 1644, so need_copy ought to return `true`
+    scene
+        .ucmd()
+        .arg("-C")
+        .arg(file1)
+        .arg(file2)
+        .succeeds()
+        .no_stderr();
+
+    file2_meta = at.metadata(file2);
+    let after_install_sticky_again = FileTime::from_last_modification_time(&file2_meta);
+
+    assert!(after_install_sticky != after_install_sticky_again);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+#[ignore] // I'm stuck; see XXX note inside.
+fn test_install_copy_then_compare_nonregular_file() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let file1 = "test_install_copy_then_compare_nonregular_file_a1";
+    let file2 = "test_install_copy_then_compare_nonregular_file_a2";
+
+    mknod(
+        file1, // XXX: How do I get this into scene.tmpd?
+        SFlag::S_IFIFO,
+        Mode::from_bits_truncate(0o644),
+        0,
+    )
+    .unwrap();
+    use std::process::Stdio;
```

Comment:
this too

Reply:
That `use` I can remove (it was part of some earlier flailing before I understood what my actual issue was), but did you see the comment on L426? I can easily get the test working once I know how to run the `mknod` within `scene.tmpd` — without actually accessing that fields, as it's private.

Reply:
[L437 now](https://github.com/uutils/coreutils/pull/1811/files#diff-28f964575c1020d23413d29689d0eca826f901cbd166445db2ef984b660540f5R437).

Reply:
@Goorzhel You could scoop the tmp path out, as a string, by using `at.plus_as_string(file1)`
[Like here.](https://github.com/uutils/coreutils/blob/master/tests/by-util/test_cp.rs#L1030)

Reply:
Awesome, that's one hurdle down. There is one immediately afterward, though: none of the `SFlag`s worked out well.
* `S_IFBLK`: `mknod` throws `EPERM`. (Solvable by granting `CAP_MKNOD`, but that requires root.)
* `S_IFCHR`: ditto.
* `S_IFDIR`: ditto, but not quite the point of this test anyway.
* `S_IFIFO`: hangs indefinitely.
* `S_IFLNK`: `mknod` throws `EPERM`.
* `S_IFSOCK`: `mknod` works fine, but the attempt to `install` is denied by:
```
openat(AT_FDCWD, "test_install_copy_then_compare_nonregular_file_a1", ...) = -1 ENXIO (No such device or address)
```